src/Raven.Server/Rachis/Leader.cs

```diff
@@ -863,10 +882,24 @@ public bool TryModifyTopology(string nodeTag, string nodeUrl, TopologyModificati
                     Interlocked.Exchange(ref _topologyModification, null);
                 });
             }
-            _hasNewTopology.Raise();
-            _voterResponded.Set();
-            _newEntry.Set();
-
+            //Need to protect ourselves from disposed events.
+            bool lockTaken = false;
+            Monitor.TryEnter(_eventsLocker, ref lockTaken);
```

Comment:
Calling `TryModifyTopology` concurrently will also cause this.
Let's use a better method, see `DisposeLock`

Reply:
i'll do that